src/main/java/org/verdictdb/coordinator/SelectQueryCoordinator.java

```diff
@@ -178,13 +178,23 @@ public void abort() {
    * @return false if the query contain the syntax that is not supported by VerdictDB
    */
   private boolean isSupportQuery(SelectQuery query) {
```

Comment:
Let's make this function throw an exception in case the query is not supportable. Let the exception include an appropriate message that explains why this query is not supported. Let's change the function name to "ensureQuerySupport".